Zend/zend_compile.c

```diff
@@ -8137,6 +8137,30 @@ void zend_compile_shell_exec(znode *result, zend_ast *ast) /* {{{ */
 }
 /* }}} */
 
+void zend_compile_array_fill(znode *result, zend_ast *ast) /* {{{ */
+{
+	zend_op *opline;
+
+	zend_ast *value_ast = ast->child[0];
+	znode value_node;
+	zend_compile_expr(&value_node, value_ast);
+
+	zval *size_zv = zend_ast_get_zval(ast->child[1]);
```

Comment:
If you want to require a known, static size, then you should probably use something more precise for $4 in the grammar.   As it stands trying to use a non-constant will trigger an assert in a debug build.

T_LNUMBER in the grammer would be the simplest, though if you want to capture constexprs like `1 + 2 + 3`, then you can leave it as expr, use zend_compile_expr() on it, and validate that the result is an IS_CONSTANT node.